presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -123,19 +129,35 @@ protected Node visitShowStats(ShowStats node, Void context)
             if (node.getRelation() instanceof TableSubquery) {
                 Query query = ((TableSubquery) node.getRelation()).getQuery();
                 QuerySpecification specification = (QuerySpecification) query.getQueryBody();
-                Plan plan = queryExplainer.get().getLogicalPlan(session, query(specification), parameters, warningCollector);
+                Plan plan = getPlanForSpecification(specification);
                 validateShowStatsSubquery(node, query, specification, plan);
                 Table table = (Table) specification.getFrom().get();
                 Constraint constraint = getConstraint(plan);
-                return rewriteShowStats(node, table, constraint);
+                return rewriteShowStats(node, table, specification.getSelect().getSelectItems(), constraint);
             }
             if (node.getRelation() instanceof Table) {
                 Table table = (Table) node.getRelation();
-                return rewriteShowStats(node, table, Constraint.alwaysTrue());
+                return rewriteShowStats(node, table, ImmutableList.of(new AllColumns()), Constraint.alwaysTrue());
             }
             throw new IllegalArgumentException("Expected either TableSubquery or Table as relation");
         }
 
+        private TableMetadata getTableMetadata(Node node, QualifiedName table)
+        {
+            return metadata.getTableMetadata(session, getTableHandle(node, table));
+        }
+
+        private Plan getPlanForSpecification(QuerySpecification specification)
+        {
+            try {
+                return queryExplainer.get().getLogicalPlan(session, query(specification), parameters, warningCollector);
+            }
+            catch (AccessDeniedException e) {
```

Comment:
In the next commit you remove this catch clause, and in the subsequent you add it back. Am i reading it right?